main.c

```diff
@@ -954,12 +957,18 @@ load_dir (struct ovl_data *lo, struct ovl_node *n, struct ovl_layer *layer, char
                 }
             }
 
+          sprintf (node_path, ".wh.%s", dent->d_name);
+          ret = TEMP_FAILURE_RETRY (fstatat (fd, node_path, &tmp_st, AT_SYMLINK_NOFOLLOW));
+          if (ret < 0 && errno != ENOENT)
+            {
+              closedir (dp);
+              return NULL;
+            }
           sprintf (node_path, "%s/%s", n->path, dent->d_name);
```

Comment:
This looks strange?  You are overriding the previous settings of node_path?

Reply:
I am just re-using the same buffer so that I don't need more space on the stack.

Reply:
OK, just checking.